AsyncDisplayKit/Details/ASRangeControllerBeta.mm

```diff
@@ -38,16 +40,49 @@ - (instancetype)init
   }
   
   _rangeIsValid = YES;
+  _currentRangeMode = ASLayoutRangeModeCount;
   
   return self;
 }
 
+- (void)dealloc
+{
+  if (_didRegisterForNotifications) {
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
```

Comment:
I have always preferred this pattern as well, rather than specifying the notification to remove. But because our app uses notifications so heavily, I have heard that this is meaningfully less efficient then removing this specific notification. Could you change it to two calls to this to specify the notification that we subscribe to?